web/settings/capabilities.go

```diff
@@ -0,0 +1,55 @@
+package settings
+
+import (
+	"net/http"
+
+	"github.com/cozy/cozy-stack/model/instance"
+	"github.com/cozy/cozy-stack/model/permission"
+	"github.com/cozy/cozy-stack/pkg/config/config"
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/jsonapi"
+	"github.com/cozy/cozy-stack/web/middlewares"
+	"github.com/labstack/echo/v4"
+)
+
+type apiCapabilities struct {
+	DocID          string `json:"_id"`
+	FileVersioning bool   `json:"file_versioning"`
+}
+
+func (c *apiCapabilities) ID() string                             { return c.DocID }
+func (c *apiCapabilities) Rev() string                            { return "" }
+func (c *apiCapabilities) DocType() string                        { return consts.Settings }
+func (c *apiCapabilities) Clone() couchdb.Doc                     { cloned := *c; return &cloned }
+func (c *apiCapabilities) SetID(id string)                        { c.DocID = id }
+func (c *apiCapabilities) SetRev(rev string)                      {}
+func (c *apiCapabilities) Relationships() jsonapi.RelationshipMap { return nil }
+func (c *apiCapabilities) Included() []jsonapi.Object             { return nil }
+func (c *apiCapabilities) Links() *jsonapi.LinksList {
+	return &jsonapi.LinksList{Self: "/settings/capabilities"}
+}
+func (c *apiCapabilities) Fetch(field string) []string { return nil }
+
+func newCapabilities(inst *instance.Instance) *apiCapabilities {
+	// File versioning is enabled for all instances, except for the Swift
+	// layout v1 and v2
+	versioning := true
+	switch config.FsURL().Scheme {
+	case config.SchemeSwift, config.SchemeSwiftSecure:
```

Comment:
For my comprehension, why a `case` and not a simple `if` ?

Reply:
With an `if` like that, it would work too:

```go
scheme := config.FsURL().Scheme
if scheme == config.SchemeSwift || scheme.SchemeSwiftSecure {
```

I've chosen the `switch case` as it is more easier to extend with a new `case`, but I don't really care.

Reply:
ok, thanks for the explanation.